Tools/scripts/summarize_stats.py

```diff
@@ -120,6 +123,26 @@ def load_raw_data(input: Path) -> RawData:
 def save_raw_data(data: RawData, json_output: TextIO):
     json.dump(data, json_output)
 
+@functools.cache
+def _get_uop_flags_from_file(
+    flag_names: tuple[str] = None,
+    filepath: str | Path = "Include/internal/pycore_uop_metadata.h",
+) -> dict[str, list[str]]:
+    flags = {}
```

Comment:
Overall, I found this function a bit hard to follow (but then again, parsing C with regexes is always that way).  It might be helpful to have a header comment with an example of the pattern it's looking for and what the output would look like for it.

Reply:
Agreed, I've added a comment to clarify. In general, I wonder if having something like `opcode_metadata_generator.py` for the UOp metadata, and spitting out that metadata as Python originally, would be more robust than parsing the C headers, but that might be a significant chunk of work.